src/arctic3d/cli.py

```diff
@@ -192,22 +205,32 @@ def main(
     init_message = get_init_message()
     log.info(init_message)
     st_time = time.time()
-    inp = Input(input_arg)
+
+    # handling input files
     input_files = {}
-    # retrieve uniprot information
-    if inp.is_fasta():
-        input_files["fasta"] = Path(inp.arg)
-        uniprot_id = run_blast(input_files["fasta"], db)
-    if inp.is_uniprot():
-        uniprot_id = inp.arg.upper()
-    if inp.is_pdb():
-        input_files["pdb"] = Path(inp.arg)
+    uniprot_id = None
+    # fasta input
+    if is_fasta(input_fasta):
+        input_files["fasta"] = Path(input_fasta)
+        if not is_uniprot(input_uniprot):
+            uniprot_id = run_blast(input_files["fasta"], db)
+        else:
+            log.warning("Uniprot ID provided. The fasta file will be ignored.")
+    # pdb input
+    if is_pdb(input_pdb):
+        input_files["pdb"] = Path(input_pdb)
         if not interface_file:
             fasta_f = to_fasta(input_files["pdb"], temp=False)
-            uniprot_id = run_blast(fasta_f.name, db)
+            if not is_uniprot(input_uniprot):
+                uniprot_id = run_blast(fasta_f.name, db)
+                # remove fasta_f
+                input_files["pdb"].with_suffix(".fasta").unlink()
         else:
             input_files["interface_file"] = Path(interface_file)
-            uniprot_id = None
+    # 25/7/2023: if the pdb is provided, the blast uniprot_id can be
+    # overwritten by specifying the uniprot_id input argument
+    if is_uniprot(input_uniprot):
+        uniprot_id = input_uniprot.upper()
```

Comment:
Since you are using only `if` and not `if/elif/else`, `uniprot_id` can be `None` by the end of the switch. Here you should either re-work this logic and add hierarchy to the parameters (with `if/elif/else`) or add an extra check that `uniprot_id` is not `None`

Reply:
but it's fine to have `uniprot_id = None` at the end of the switch!